File: clojure/anagram/src/anagram.clj
(ns anagram)

(defn- anagram? [w1 w2]
  (let [w1' (.toUpperCase w1)
        w2' (.toUpperCase w2)]
    (and
     (not= w1' w2')
     (= (sort w1') (sort w2')))))

(defn anagrams-for [word wlist]
  (filter (partial anagram? word) wlist))

File: haskell/anagram/src/Anagram.hs
module Anagram (anagramsFor) where

import Data.List (sort)
import Data.Char (toLower)

isAnagram :: String -> String -> Bool
isAnagram w a
  | w' == a'  = False
  | otherwise = sort w' == sort a'
  where w' = map toLower w
        a' = map toLower a

anagramsFor :: String -> [String] -> [String]
anagramsFor xs = filter (isAnagram xs)
